testsuite/tests/typing-unboxed/external_unboxed.ml

```diff
@@ -0,0 +1,40 @@
+(* TEST
+   flags = "-strict-sequence"
+   * expect
+*)
+
+(* MPR#7828 *)
+type i = I of int
+external id : i -> i = "%identity";;
+[%%expect{|
+type i = I of int
+Line 2, characters 0-34:
+  external id : i -> i = "%identity";;
+  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Warning 61: This primitive declaration uses type i, which is unannotated and
+unboxable. The representation of such types may change in future
+versions. You should annotate the declaration of i with [@@boxed]
+or [@@unboxed].
+Line 2, characters 0-34:
+  external id : i -> i = "%identity";;
+  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Warning 61: This primitive declaration uses type i, which is unannotated and
+unboxable. The representation of such types may change in future
+versions. You should annotate the declaration of i with [@@boxed]
+or [@@unboxed].
```

Comment:
Why is this warning printed twice for the same location? Is it because it is printed once for each occurrence of `i`? Is it easy to fix the location of the warning to be on the use of `i` (so two different locations), instead of being on the whole external declaration? (cc @Armael, who is interested in multi-location compiler messages.)

Reply:
My idea was to have two warnings shown, one with both the left `i`'s location and the whole external location, and the other with the right `i`'s location and the whole external location.